src/NetworkEquipmentModelStencil.php

```diff
@@ -0,0 +1,111 @@
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+use Glpi\Application\View\TemplateRenderer;
+
+class NetworkEquipmentModelStencil extends Stencil
+{
+    public static function getTypeName($nb = 0): string
+    {
+        return __('Graphical slot definition');
+    }
+
+    public function getPicturesFields(bool $editor): array
+    {
+        return ['picture_front', 'picture_rear'];
+    }
+
+    public function getParams(bool $editor): array
+    {
+        if ($editor) {
+            return [
+                'nb_zones_label' => __('Set number of ports'),
+                'define_zones_label' => __('Define port data in image'),
+                'zone_label' => __('Port Label'),
+                'zone_number_label' => __('Port Number'),
+                'save_zone_data_label' => __('Save port data'),
+                'add_zone_label' => __('Add a new port'),
+                'remove_zone_label' => __('Remove last port'),
+            ];
+        } else {
+            return [
+                'anchor_id' => 'port_number_',
+            ];
+        }
+    }
+
+    public static function displayTabContentForItem(CommonGLPI $item, $tabnum = 1, $withtemplate = 0)
+    {
+        $stencil = Stencil::getStencilFromItem($item);
+        if ($stencil != null) {
+            $stencil->displayStencilEditor();
+            return true;
+        }
+
+        return false;
+    }
+
+    public function getZoneLabel(bool $editor, array $port)
+    {
+        return parent::getZoneLabel($editor, $port) . (!$editor ? TemplateRenderer::getInstance()->render('stencil/parts/port/status.html.twig', [
+            'port' => self::getPortInformation($port),
+        ]) : '');
+    }
+
+    public function getZonePopover(bool $editor, array $port)
+    {
+        return parent::getZonePopover($editor, $port) . (!$editor ? TemplateRenderer::getInstance()->render('stencil/parts/port/popover.html.twig', [
+            'port' => self::getPortInformation($port),
+        ]) : '');
+    }
+
+    private function getPortInformation(array $port): array
+    {
+        $networkPort = new NetworkPort();
+
+        if (
+            $networkPort->getFromDBByCrit([
+                'logical_number' => $port['number'],
+                'items_id'  => $this->getStencilItem()->getID(),
```

Comment:
`getStencilItem` return type is `CommonGLPI`, which does not have the `getID()` method.
Maybe the return type should be `CommondDBTM` instead ?